core/src/main/java/software/amazon/awssdk/core/auth/AwsCredentialsProviderChain.java

```diff
@@ -103,10 +103,7 @@ public AwsCredentials getCredentials() {
 
     @Override
     public void close() {
-        credentialsProviders.stream()
-                            .filter(AutoCloseable.class::isInstance)
-                            .map(AutoCloseable.class::cast)
-                            .forEach(c -> IoUtils.closeQuietly(c, null));
+        credentialsProviders.forEach(c -> IoUtils.closeIfCloseable(c, null));
```

Comment:
I feel like `IoUtils` should expose a `closeQuietly` / `closeIfClosable` that doesn't require a log to be passed - ie to the overload in the `Utils` itself. Then you could do this via method reference and probably in other places too.

Reply:
I'd like that, but I'd guess that would mean no one would ever use the current one. Not that we use it much now, but at least now we're consciously saying "yeah, I'm not going to use it".

It might be less annoying to use if we also support passing in a Class<?> instead of a logger, and completely disallow null?